runtime/lib/binary.cpp

```diff
@@ -138,7 +138,8 @@ std::vector<TensorDesc> getProgramInputs(Flatbuffer binary,
                                          std::uint32_t programIndex) {
   std::vector<TensorDesc> inputs;
   auto const *program = getBinary(binary)->programs()->Get(programIndex);
-  for (auto const *input : *program->inputs()) {
+  assert(program->device_programs()->size() == 1);
```

Comment:
Yes that's right, we only support a single device today.  The top level program has potentially many underlying per-device programs, but since we only support single device today there is only 1 device program expected.